system/security/digisign/digisign.cpp

```diff
@@ -0,0 +1,270 @@
+        return false;
+#endif
+    }
+
+
+    //Verify the given text was used to create the given digital signature
+    bool digiVerify(const char * text, StringBuffer & b64Signature)
+    {
+        if (!verifyingConfigured)
+            throw MakeStringException(-1, "digiVerify:Verifying Digital Signatures not configured");
+
+        CriticalBlock b(digiVerifyCrit);
+
+        if (publicKeyBuff.isEmpty())
+            publicKeyBuff.loadFile(publicKeyFile.str());
+        if (publicKeyBuff.isEmpty())
+            throw MakeStringException(-1, "digiSign:Cannot load public key file");
+
+#ifdef _USE_ZLIB
+        BIO * keybio = BIO_new_mem_buf((void*) publicKeyBuff.str(), -1);
+        if (keybio == nullptr)
+        {
+            char buff[120];
+            ERR_error_string(ERR_get_error(), buff);
+            throw MakeStringException(-1, "digiVerify:BIO_new_mem_buf: %s", buff);
+        }
+        RSA * rsa = PEM_read_bio_RSA_PUBKEY(keybio, nullptr, nullptr, nullptr);
+
+        EVP_PKEY* pubKey = EVP_PKEY_new();
+        EVP_PKEY_assign_RSA(pubKey, rsa);
+        EVP_MD_CTX * RSAVerifyCtx = EVP_MD_CTX_create();//allocate, initializes and return a digest context
+        if (RSAVerifyCtx == nullptr)
+        {
+            EVP_PKEY_free(pubKey);
+            BIO_free_all(keybio);
+            char buff[120];
+            ERR_error_string(ERR_get_error(), buff);
+            throw MakeStringException(-1, "digiVerify:EVP_MD_CTX_create: %s", buff);
+        }
+
+        if (EVP_DigestVerifyInit(RSAVerifyCtx, nullptr, EVP_sha256(), nullptr, pubKey) <= 0)
+        {
+            EVP_PKEY_free(pubKey);
+            EVP_MD_CTX_destroy(RSAVerifyCtx);//cleans up digest context ctx
+            BIO_free_all(keybio);
+            char buff[120];
+            ERR_error_string(ERR_get_error(), buff);
+            throw MakeStringException(-1, "digiVerify:EVP_DigestVerifyInit: %s", buff);
+        }
+        BIO_free_all(keybio);
+
+        //decode base64 signature
+        StringBuffer decodedSig;
+        JBASE64_Decode(b64Signature.str(), decodedSig);
+        if (EVP_DigestVerifyUpdate(RSAVerifyCtx, text, strlen(text)) <= 0)
+        {
+            EVP_PKEY_free(pubKey);
+            EVP_MD_CTX_destroy(RSAVerifyCtx);//cleans up digest context ctx
+            char buff[120];
+            ERR_error_string(ERR_get_error(), buff);
+            throw MakeStringException(-1, "digiVerify:EVP_DigestVerifyUpdate: %s", buff);
+        }
+
+        int match = EVP_DigestVerifyFinal(RSAVerifyCtx, (unsigned char *)decodedSig.str(), decodedSig.length());
+        EVP_PKEY_free(pubKey);
+        EVP_MD_CTX_destroy(RSAVerifyCtx);//cleans up digest context ctx
+        return match == 1;
+#else
+        throw MakeStringException(-1, "digiSign:Platform built without ZLIB");
+        return false;
+#endif
+    }
+} CDigitalSignatureManager;
+
+
+extern "C"
+{
+
+    DIGISIGN_API IDigitalSignatureManager * digitalSignatureManagerInstance()
```

Comment:
Do you have an example of how this is called/used ?
Should a test be added ?

Reply:
if (digitalSignatureManagerInstance()->isDigiSignerConfigured() )
    digitalSignatureManagerInstance()->digiSign("Hello World", sbSignature);

Reply:
As we discussed , tests should be added to regression suite, via separate effort